### ISSUE TITLE
Unresponsive modal when RefreshControl used with iOS modal and header hidden

### ISSUE THREAD

**Issue by chvanlennep** [react-navigation/react-navigation#12355]
### Current behavior

When displaying a native iOS modal containing a `RefreshControl` component in a scrollable component, if `headerShown: false` is set on the screen options, the modal is unresponsive on mount. Backgrounding the app and returning fixes the issue.

### Expected behavior

The iOS modal in this setup should be responsive on mount.

### Reproduction

https://github.com/chvanlennep/RefreshControlExample

### Platform

- [ ] Android
- [x] iOS
- [ ] Web
- [ ] Windows
- [ ] MacOS

### Packages

- [ ] @react-navigation/bottom-tabs
- [ ] @react-navigation/drawer
- [ ] @react-navigation/material-top-tabs
- [ ] @react-navigation/stack
- [x] @react-navigation/native-stack
- [ ] react-native-drawer-layout
- [ ] react-native-tab-view

### Environment

- [x] I've removed the packages that I don't use

| package                                | version |
| -------------------------------------- | ------- |
| @react-navigation/native               | 7.0.14
| @react-navigation/bottom-tabs          | 7.2.0
| @react-navigation/native-stack         |^7.1.13
| react-native-screens                   |~4.1.0
| react-native-safe-area-context         |4.12.0
| react-native-gesture-handler           |~2.20.2
| react-native                           |0.76.3
| expo                                   |~52.0.11
| node                                   |18.18.0
| npm or yarn                            |1.22.22

**Comment by chvanlennep** [comment]
> The versions mentioned in the issue for the following packages differ from the latest versions on npm:
> 
> * `@react-navigation/native` (found: `7.0.12`, latest: `7.0.14`)
> * `@react-navigation/bottom-tabs` (found: `7.1.1`, latest: `7.2.0`)
> 
> Can you verify that the issue still exists after upgrading to the latest versions of these packages?

Yes, I have updated the issue and the repo with these latest versions as well.

**Comment by chvanlennep** [comment]
> Hey! Thanks for opening the issue. Seems that this issue is related to `react-native-screens` library which is a dependency of React Navigation. Can you also post your issue in [this repo](https://github.com/software-mansion/react-native-screens) so that it's notified to the maintainers of that library? This will help us fix the issue faster since it's upto the maintainers of that library to investigate it.

Done:

- [ ] https://github.com/software-mansion/react-native-screens/issues/2586

**Comment by kkafar** [comment]
Moved to react-native repo: https://github.com/facebook/react-native/issues/48579
with fix suggestion: https://github.com/facebook/react-native/pull/48580
